R/target_market_share.R

```diff
@@ -80,12 +108,28 @@ target_market_share <- function(data,
       ald,
       scenario,
       region_isos
-    ) %>%
-    summarize_weighted_production(
-      !!!rlang::syms(summary_groups),
-      use_credit_limit = use_credit_limit
-    ) %>%
-    add_ald_benchmark(ald, region_isos, by_company)
+    )
+
+  if (weight_production) {
+    data <- data %>%
+      summarize_weighted_production(
+        !!!rlang::syms(summary_groups),
+        use_credit_limit = use_credit_limit
+        ) %>%
+      add_ald_benchmark(ald, region_isos, by_company)
+  } else {
+    data <- data %>%
+      group_by(
+        .data$sector_ald,
+        .data$technology,
+        .data$year,
+        !!!rlang::syms(summary_groups)
+        ) %>%
+      summarize(
+        weighted_production = sum(.data$production) # this is not actually "weighted"
```

Comment:
Maybe we can capture the comment on a helper, then extract the common pattern in both branches of the if else.